app/scripts/modules/amazon/src/serverGroup/configure/wizard/instanceType/CpuCreditsToggle.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import { ToggleButtonGroupInput } from '@spinnaker/core';
+
+import { AwsReactInjector } from 'amazon/reactShims';
+import { IAmazonServerGroupCommand } from '../../serverGroupConfiguration.service';
+
+export interface ICpuCreditsToggleProps {
+  command: IAmazonServerGroupCommand;
+  newInstanceType?: string;
+  newProfileType?: string;
+  setUnlimitedCpuCredits: (unlimitedCpuCredits: boolean | undefined) => void;
+}
+
+export class CpuCreditsToggle extends React.Component<ICpuCreditsToggleProps> {
+  private handleInstanceTypeChange() {
+    const isBurstingSupported = AwsReactInjector.awsInstanceTypeService.isBurstingSupported(this.props.newInstanceType);
+    if (isBurstingSupported) {
+      return true;
+    }
+    this.props.setUnlimitedCpuCredits(undefined);
+    return false;
+  }
+
+  private handleProfileChange() {
+    const instanceType = this.props.command.instanceType;
+    const isTypeInProfile = AwsReactInjector.awsInstanceTypeService.isInstanceTypeInCategory(
+      instanceType,
+      this.props.newProfileType,
```

Comment:
What is the difference between the `command.instanceType` and the `newInstanceType`?

Reply:
`newInstanceType` is the state maintained in `CpuCreditsToggle's` parent in order to correctly identify an instance type change and differentiate it from a profile change. Replacing `newInstancetype` with `command.instanceType` will cause the `handleInstanceTypeChange` code path to be executed when instance type didn't change i.e. during profile changes.